hal/network/ncp_client/sara/sara_ncp_client.cpp

```diff
@@ -1377,11 +1383,20 @@ int SaraNcpClient::registerNet() {
     connectionState(NcpConnectionState::CONNECTING);
     registeredTime_ = 0;
 
-    CHECK_PARSER_OK(parser_.execCommand("AT+COPS?"));
+    auto resp = parser_.sendCommand("AT+COPS?");
+    int copsState = -1;
+    r = CHECK_PARSER(resp.scanf("+COPS: %d", &copsState));
+    CHECK_TRUE(r == 1, SYSTEM_ERROR_AT_RESPONSE_UNEXPECTED);
+    r = CHECK_PARSER(resp.readResult());
+    CHECK_TRUE(r == AtResponse::OK, SYSTEM_ERROR_AT_NOT_OK);
 
     // NOTE: up to 3 mins (FIXME: there seems to be a bug where this timeout of 3 minutes
     //       is not being respected by u-blox modems.  Setting to 5 for now.)
-    r = CHECK_PARSER(parser_.execCommand(5 * 60 * 1000, "AT+COPS=0,2"));
+    if (copsState != 0) {
+        // If the set command with <mode>=0 is issued, a further set
+        // command with <mode>=0 is managed as a user reselection
+        r = CHECK_PARSER(parser_.execCommand(5 * 60 * 1000, "AT+COPS=0,2"));
+    }
```

Comment:
One quick thought on this: As written in the attached AT manual, this will trigger a re-selection which we want to avoid for faster bootup times (which is all good). If we have to deal with roaming locations though, is it worth having an `AT+COPS=0` setting to in fact trigger a re-selection because otherwise what if it doesn't attach to network as expected. Am I making sense? Should we have a check that tells when to set `AT+COPS=0` and when not to set it? I know we have 10 min timeout, but even after 10min when we hit this line, looks like we won't do another `AT+COPS=0` in case which we may not register on a given network with edge case roaming scenarios and such?

Reply:
But after 10 mins and a modem reset the modem on its own will start a new registration process (auto-COPS), right?

Reply:
Hmm.. good point. That's also right. It should. (Reset is supposed to be more powerful than anything else.)